<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View3D</class>
 <widget class="QMainWindow" name="View3D">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D View</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/icons/icons/package.png</normaloff>:/icons/icons/package.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Alignment Parameters</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>X</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="x">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Y</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="y">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Z</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="z">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Phi</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="phi">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Theta</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="theta">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Omega</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="omega">
             <property name="minimum">
              <double>-10000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>10000.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Dataset</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Run ID</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="run_ids"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Event Filter</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QComboBox" name="eventFilter">
             <item>
              <property name="text">
               <string>All Events</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Fiducial Hits</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Fiducial Misses</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="viewLayout">
        <item>
         <widget class="Viewport" name="view">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QDoubleSpinBox" name="timedelta">
        <property name="suffix">
         <string>s</string>
        </property>
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="playback">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/icons/icons/control_play.png</normaloff>
          <normalon>:/icons/icons/control_pause.png</normalon>:/icons/icons/control_play.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="event">
        <property name="prefix">
         <string>Event </string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="eventSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="numEvents">
        <property name="text">
         <string>000</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="closeButton">
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Viewport</class>
   <extends>QOpenGLWidget</extends>
   <header>viewport.h</header>
   <slots>
    <signal>numEventsChanged(int)</signal>
    <slot>setParameterX(double)</slot>
    <slot>setParameterY(double)</slot>
    <slot>setParameterZ(double)</slot>
    <slot>setParameterPhi(double)</slot>
    <slot>setParameterTheta(double)</slot>
    <slot>setParameterOmega(double)</slot>
    <slot>setEventIndex(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>x</tabstop>
  <tabstop>y</tabstop>
  <tabstop>z</tabstop>
  <tabstop>phi</tabstop>
  <tabstop>theta</tabstop>
  <tabstop>omega</tabstop>
  <tabstop>run_ids</tabstop>
  <tabstop>timedelta</tabstop>
  <tabstop>playback</tabstop>
  <tabstop>event</tabstop>
  <tabstop>eventSlider</tabstop>
  <tabstop>closeButton</tabstop>
 </tabstops>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>eventSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>event</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>580</y>
    </hint>
    <hint type="destinationlabel">
     <x>194</x>
     <y>580</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>event</sender>
   <signal>valueChanged(int)</signal>
   <receiver>eventSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>184</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>575</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>View3D</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>749</x>
     <y>577</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>x</sender>
   <signal>valueChanged(double)</signal>
   <receiver>view</receiver>
   <slot>setParameterX(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>266</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>y</sender>
   <signal>valueChanged(double)</signal>
   <receiver>view</receiver>
   <slot>setParameterY(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>z</sender>
   <signal>valueChanged(double)</signal>
   <receiver>view</receiver>
   <slot>setParameterZ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>phi</sender>
   <signal>valueChanged(double)</signal>
   <receiver>view</receiver>
   <slot>setParameterPhi(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>118</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>138</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>theta</sender>
   <signal>valueChanged(double)</signal>
   <receiver>view</receiver>
   <slot>setParameterTheta(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>omega</sender>
   <signal>valueChanged(double)</signal>
   <receiver>view</receiver>
   <slot>setParameterOmega(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>229</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>run_ids</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>View3D</receiver>
   <slot>selectRunIndex(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>265</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>event</sender>
   <signal>valueChanged(int)</signal>
   <receiver>view</receiver>
   <slot>setEventIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>580</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>408</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view</sender>
   <signal>numEventsChanged(int)</signal>
   <receiver>View3D</receiver>
   <slot>setNumEvents(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>486</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>selectRunIndex(QString)</slot>
  <slot>setNumEvents(int)</slot>
 </slots>
</ui>
